Sources/PenguinParallel/ConcurrencyPlatform.swift

```diff
@@ -146,6 +146,8 @@ public protocol RawThreadLocalStorage {
 
   /// Makes a new key; the returned key should be used for the entire process lifetime.
   static func makeKey() -> Key
+  /// Invalidates a previously constructed key, freeing resources
+  static func destroyKey(_ key: inout Key)
```

Comment:
Would a better design be for the `Key`'s destructor to handle cleanup?

Reply:
The `destroyKey` is different from the key destructor - it frees the TLS slot for re-use, it does not handle the cleanup of the TLS key itself, which remains the duty of the TLS destructor.